packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -190,11 +218,28 @@ async function loadConnections(
 ) {
   try {
     const loadedConnections = await connectionStorage.loadAll();
+    const toBeReSaved: ConnectionInfo[] = [];
```

Comment:
Not blocking. There is something odd to me with the fact that the frontend doesn't directly use the tokens, and could operate without its knowledge .. but here it seems to be the thing that orchestrates storing it. 

Maybe a bit of a stretch and unnecessary, but did we consider having a separate store for tokens and/or push this logic out of the frontend?

Reply:
> Not blocking. There is something odd to me with the fact that the frontend doesn't directly use the tokens, and could operate without this knowledge .. but here it seems to be the thing that orchestrates storing it.

Sooo … that’s part of the idea behind what you commented on above – this part of the code _should_ operate with as little knowledge about what’s happening here under the hood.

> Maybe a bit of a stretch and unnecessary, but did we consider having a separate store for tokens and/or push this logic out of the frontend?

I think the part that I’m having a hard time understanding here is where the boundaries of “frontend” would be, or where a better place for this logic would be.

It would be great to refactor this code to have a layer between `useConnections()`, which should really only be a React store, and `ConnectionStorage`, which should really only provide primitives for loading/saving connection info objects.

Right now, that layer doesn’t really exist, though, and I’m not sure how much work adding it would be here. If we think it’s worth it, we could make this part of this project.

Reply:
Yeah, no worries, feel free to ignore it. I'd love the idea of `useConnections` but let's keep it for later. 

I was thinking more about moving it out of the redux cycle, and avoid the generic merge:

```
ds = new Dataservice(storeOidcToken = () => {
   ConnectionStorage.updateOidcToken(connectionId, token)
});
```

But is very dependent on my read of all the moving parts, i see `ConnectionStorage` as a repository for domain objects, which means that it is correct for it to have understanding of domain concepts like oidc, and not just a crud thing.

> should operate with as little knowledge about what’s happening here under the hood.

We are generalising what’s happening here under the hood so it doesn't feel like it, but we are still doing it .. and there is nothing wrong with it, especially if is necessary or the best practical approach.

Reply:
> I was thinking more about moving it out of the redux cycle, and avoid the generic merge:

To be clear, I like that approach and I would prefer it, but unfortunately, the compass-connections package assumes that for the lifetime of the Compass process, _it_ is the thing that maintains the state of the connection storage (because it essentially keeps another, in-memory copy of that). We _should_ get rid of that, and move the actual connection "collection management" logic out of it – it’s just not quite straightforward to do.

I’ve opened https://jira.mongodb.org/browse/COMPASS-6909 as a tech debt ticket to track that work, and also listed another example of where this is already problematic (namely, that after importing connections, the import logic has to call the hook’s expose `reloadConnections` method which is being passed through the actual frontend).

Reply:
Awesome, thanks! We should try to prioritise that soon